en/computer_vision/visual_inertial_odometry.md

```diff
@@ -0,0 +1,177 @@
+- [Verify that VIO is Setup Correctly](#verify_estimate) before your first flight!
+
+
+
+### PX4 Tuning {#ekf2_tuning}
+
+The following parameters must be set to use external position information with EKF2.
+
+Parameter | Setting for External Position Estimation
+--- | ---
+[EKF2_AID_MASK](../advanced_config/parameter_reference.md#EKF2_AID_MASK) | Set *vision position fusion*, *vision velocity fusion*, *vision yaw fusion* and *external vision rotation* accoring to your desired fusion model.
+[EKF2_HGT_MODE](../advanced_config/parameter_reference.md#EKF2_HGT_MODE) | Set to *Vision* to use the vision a primary source for altitude estimation.
+[EKF2_EV_DELAY](../advanced_config/parameter_reference.md#EKF2_EV_DELAY) | Set to the difference between the timestamp of the measurement and the "actual" capture time. For more information see [below](#tuning-EKF2_EV_DELAY).
+[EKF2_EV_POS_X](../advanced_config/parameter_reference.md#EKF2_EV_POS_X), [EKF2_EV_POS_Y](../advanced_config/parameter_reference.md#EKF2_EV_POS_Y), [EKF2_EV_POS_Z](../advanced_config/parameter_reference.md#EKF2_EV_POS_Z) | Set the position of the vision sensor with respect to the vehicles body frame.
+
+These can be set in *QGroundControl* > **Vehicle Setup > Parameters > EKF2** (remember to reboot the flight controller in order for parameter changes to take effect).
+
+For more detailed/additional information, see: [ECL/EKF Overview & Tuning > External Vision System](../advanced_config/tuning_the_ecl_ekf.md#external-vision-system).
+
+
+
+#### Tuning EKF2_EV_DELAY {#tuning-EKF2_EV_DELAY}
+
+[EKF2_EV_DELAY](../advanced_config/parameter_reference.md#EKF2_EV_DELAY) is the *Vision Position Estimator delay relative to IMU measurements*.
+In other words, it is the difference between the vision system timestamp and the "actual" capture time that would have been recorded by the IMU clock (the "base clock" for EKF2).
+
+Technically this can be set to 0 if there is correct timestamping (not just arrival time) and timesync (e.g NTP) between MoCap and (for example) ROS computers.
+In reality, this may need some empirical tuning becuase delays in the communication chain are very setup-specific.
+It is rare that a system is setup with an entirely synchronised chain!
+
+A rough estimate of the delay can be obtained from logs by checking the offset between IMU rates and the EV rates:
+
+![ekf2_ev_delay log](../../assets/ekf2/ekf2_ev_delay_tuning.png)
+
+> **Note** A plot of external data vs. onboard estimate (as above) can be generated using [FlightPlot](../log/flight_log_analysis.md#flightplot) or similar flight analysis tools.
+
+The value can further be tuned by varying the parameter to find the value that yields the lowest EKF innovations during dynamic maneuvers.
+
+
+## Check/Verify VIO Estimate {#verify_estimate}
```

Comment:
@baumanta I copied this section from the [devguide here](https://dev.px4.io/master/en/ros/external_position_estimation.html). I believe that it is good advice for testing that the setup is good. My only concern is that original doc mixed mocap and VIO, so I worry this might not be true.

Can you also confirm that VIO streams ODOMETRY messages or VISION_POSITION_ESTIMATE or both? I think likely ODOMETRY  as these include velocities. 

Otherwise I think this is good enough. OK by you?

Reply:
Actually I have no idea... I never used VIO so I'm not sure what messages are used, sorry